src/common/reducers/snaps.js

```diff
@@ -51,7 +52,7 @@ export function snaps(state = {
         snaps: [
           ...action.payload.response.payload.snaps
         ],
-        ids: action.payload.response.result,
+        ids: union(state.ids, action.payload.response.result),
```

Comment:
With `union` and without an action to delete a snap, and subsequently remove it from the snap this list, the snap index will never update on remove, nor dependent selectors.